cmd/clusterctl/hack/create-local-repository.py

```diff
@@ -80,6 +80,12 @@
           'type': 'RuntimeExtensionProvider',
           'configFolder': 'test/extension/config/default',
       },
+      'add-on-test': {
+          'componentsFile': 'add-on-components.yaml',
+          'nextVersion': '...',
+          'type': 'AddonProvider',
+          'configFolder': '...',
+      },
```

Comment:
This seems incomplete, also, not sure about the implications of this change, what about people don't need the addon provider in their dev workflow

Reply:
Just to clarify, is this script just used for Tilt development or is it a part of any clusterctl e2e tests/validation? If it's just development then I agree we can leave it out. My main goal was to add some validation to ensure the add-on provider is working with clusterctl the way we'd expect.

Reply:
This is primarily used for helping with local development of clusterctl. 
For e2e test/validation we have a helper function in the e2e framework to create a local repository (Ref: https://github.com/ykakarap/cluster-api/blob/e6417381dd34017e683278c632ecddcd6817de11/test/framework/clusterctl/repository.go#L83).